lib/chai/core/assertions.js

```diff
@@ -215,6 +215,16 @@ module.exports = function (chai, _) {
   Assertion.addChainableMethod('contains', include, includeChainingBehavior);
   Assertion.addChainableMethod('includes', include, includeChainingBehavior);
 
+  // TODO: Add jsdoc
+
+  function assertFail (msg) {
+    if (msg) flag(this, 'message', msg);
+    var obj = flag(this, 'object');
+    throw new chai.AssertionError(msg, {}, obj);
+  }
```

Comment:
Another point to make here; for API consistency I think `should.fail` and `expect.fail` should work exactly the same as `assert.fail`. Method signature and all.